test/unit/test_subscriber.py

```diff
@@ -11,12 +13,35 @@
 
 class TestSubscriber(TestCase):
     case_id = 'e079cea4-1447-4529-aa70-8757f1806f60'
+    created = '2008-08-24T00:00:00Z'
     gcp_bucket = 'test-bucket'
     gcp_object_id = 'test-object'
     subscriber_future = 'test-future'
     subscription_path = 'test-subscription-path'
     test_data = 'test-data'
 
+    def assertLogLine(self, watcher, event, **kwargs):
+        """
+        Helper method for asserting the contents of structlog records caught by self.assertLogs.
+        Fails if no match is found. A match is based on the main log message (event) and all additional
+        items passed in kwargs.
+        :param watcher: context manager returned by `with self.assertLogs(LOGGER, LEVEL)`
+        :param event: event logged; use empty string to ignore or no message
+        :param kwargs: other structlog key value pairs to assert for
+        """
+        for record in watcher.records:
+            message_json = json.loads(record.message)
+            try:
+                if (
+                        event in message_json.get('event', '')
+                        and all(message_json[key] == val for key, val in kwargs.items())
+                ):
+                    break
+            except KeyError:
+                pass
```

Comment:
Can this record which key caused an error

except KeyError as e:
     missing_key = e.args[0]
     # record this in fail?

Reply:
Can log out missing keys but values are a little more difficult as they're obscured inside the `all()`.

Reply:
key should be fine